server/schedulers/balance_region.go

```diff
@@ -165,104 +170,128 @@ func (s *balanceRegionScheduler) Schedule(cluster schedule.Cluster, dryRun bool)
 		baseRegionFilters = append(baseRegionFilters, filter.NewRegionEmptyFilter(cluster))
 	}
 
-	for _, plan.source = range stores {
-		retryLimit := s.retryQuota.GetLimit(plan.source)
+	basePlan.step++
```

Comment:
Can we unify the usage for solver and plan? in `transferPeer`, it is solver.step++/--